Python/Product/Common/Infrastructure/ProcessOutput.cs

```diff
@@ -716,6 +716,26 @@ sealed class ProcessOutput : IDisposable {
             return true;
         }
 
+        private async Task<int> WaitAsync() {
+            return await this;
+        }
+
+        public async Task<int> WaitAsync(CancellationToken ct, bool killProcessOnCancel) {
```

Comment:
If you want to add cancellation handling for a task that you can't cancel explicitly, it is enough to use `ContinueWith`:
```cs
Foo(CancellationToken ct) => Foo().ContinueWith(t => t.GetAwaiter().GetResult(), ct);
```
And if some action is required on cancellation:
```cs
try {
    await Foo().ContinueWith(t => t.GetAwaiter().GetResult(), ct);
} catch (OperationCanceledException) when (ct.IsCancellationRequested) {
    SomeAction();
    throw;
}
```

But that isn't the case here. We have an `Exited` event, so we can either use `EventTaskSource` (it is now defined only in tests), or do the thing manually:
```cs
var tcs = new TaskCompletionSource<int>();
Exited += (o, e) => tcs.TrySetResult(0);
try {
    if (!_process.HasExited) {
        tcs.RegisterForCancellation(ct).UnregisterOnCompletion(tcs.Task);
        await tcs.Task;        
    }
    return _process.ExitCode;
} catch (OperationCanceledException) when (ct.IsCancellationRequested) {
    if (killProcessOnCancel) {
        Kill();
    }
    throw;
}
```
and `GetAwaiter` should use `WaitAsync`, not visa versa.

Reply:
Thanks! I've undone my changes to `ProcessOutput` and put this code where I needed it in `CondaEnvironmentManager`.